File: Data Structures & Algorithms/maximum-subarray/submission-4.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsum, cursum= nums[0],0
        for num in nums:
            if cursum<0:
                cursum=0
            cursum+=num
            maxsum=max(maxsum,cursum)
        return maxsum
                

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let set=new Set(nums);
        let count=0;
        for(let value of set){
            if(!set.has(value-1)){
                let length=1;
                while(set.has(value+length)){
                   length++; 
                }
                count=Math.max(length,count);
            }
        }
        return count
    }
}
